chains/lighter/message/lighter.go

```diff
@@ -0,0 +1,200 @@
+}
+
+type TxFetcher interface {
+	GetTx(hash string) (*lighter.LighterTx, error)
+}
+
+type LighterMessageHandler struct {
+	coordinator Coordinator
+	host        host.Host
+	comm        comm.Communication
+	fetcher     signing.SaveDataFetcher
+	sigChn      chan any
+
+	lighterAddress common.Address
+	usdcAddress    common.Address
+	txFetcher      TxFetcher
+}
+
+func NewLighterMessageHandler(
+	lighterAddress common.Address,
+	usdcAddress common.Address,
+	txFetcher TxFetcher,
+	coordinator Coordinator,
+	host host.Host,
+	comm comm.Communication,
+	fetcher signing.SaveDataFetcher,
+	sigChn chan any,
+) *LighterMessageHandler {
+	return &LighterMessageHandler{
+		txFetcher:      txFetcher,
+		usdcAddress:    usdcAddress,
+		lighterAddress: lighterAddress,
+		coordinator:    coordinator,
+		host:           host,
+		comm:           comm,
+		fetcher:        fetcher,
+		sigChn:         sigChn,
+	}
+}
+
+// HandleMessage finds the Mayan deposit with the according deposit ID and starts
+// the MPC signature process for it. The result will be saved into the signature
+// cache through the result channel.
+func (h *LighterMessageHandler) HandleMessage(m *message.Message) (*proposal.Proposal, error) {
+	data := m.Data.(*LighterData)
+
+	err := h.notify(data)
+	if err != nil {
+		log.Warn().Msgf("Failed to notify relayers because of %s", err)
+	}
+
+	tx, err := h.txFetcher.GetTx(data.OrderHash)
+	if err != nil {
+		data.ErrChn <- err
+		return nil, err
+	}
+
+	if err = h.verifyWithdrawal(tx); err != nil {
+		data.ErrChn <- err
+		return nil, err
+	}
+
+	data.ErrChn <- nil
+
+	calldata, err := h.calldata(tx)
+	if err != nil {
+		return nil, err
+	}
+
+	unlockHash, err := signature.BorrowUnlockHash(
+		calldata,
+		new(big.Int).SetUint64(tx.Transfer.USDCAmount),
+		h.usdcAddress,
+		ARBITRUM_CHAIN_ID,
+		h.lighterAddress,
+		//nolint:gosec
+		uint64(time.Now().Add(FILL_DEADLINE).Unix()),
+		data.Caller,
+		data.LiquidityPool,
+		data.Nonce)
+	if err != nil {
+		data.ErrChn <- err
+		return nil, err
+	}
+
+	sessionID := fmt.Sprintf("lighter-%s", "")
+	signing, err := signing.NewSigning(
+		new(big.Int).SetBytes(unlockHash),
+		sessionID,
+		sessionID,
+		h.host,
+		h.comm,
+		h.fetcher)
+	if err != nil {
+		return nil, err
+	}
+
+	err = h.coordinator.Execute(context.Background(), []tss.TssProcess{signing}, h.sigChn, data.Coordinator)
+	if err != nil {
+		return nil, err
+	}
+	return nil, nil
+}
+
+func (h *LighterMessageHandler) verifyWithdrawal(tx *lighter.LighterTx) error {
+	if tx.Type != lighter.TxTypeL2Transfer {
+		return errors.New("invalid transaction type")
+	}
+
+	if tx.Transfer.ToAccountIndex != WITHDRAWAL_ACCOUNT_INDEX {
+		return errors.New("transfer account index invalid")
+	}
+
+	return nil
+}
+
+func (h *LighterMessageHandler) calldata(tx *lighter.LighterTx) ([]byte, error) {
+	return consts.LighterABI.Pack(
+		"withdraw",
+		common.HexToHash(tx.Hash),
+		common.HexToAddress(tx.L1Address),
+		new(big.Int).SetUint64(tx.Transfer.USDCAmount))
```

Comment:
I think you should use amount from liquidity request. This is amount with fee and we should authorize only fill without fee.